astroid/bases.py

```diff
@@ -146,9 +149,12 @@ def _infer_stmts(
 ) -> collections.abc.Generator[InferenceResult, None, None]:
     """Return an iterator on statements inferred by each statement in *stmts*."""
     inferred = False
+    constraint_failed = False
+    constraints: dict[str, dict[nodes.If, Constraint]] = {}
     if context is not None:
         name = context.lookupname
         context = context.clone()
+        constraints = context.constraints.get(name, {})
```

Comment:
It's better to avoid assigning a different type to an existing variable. That will mess with type checking and make it more difficult to understand. Better to choose a unique name for it.

Reply:
I think it's fixed in the latest version ``context.constraints`` return a dict.